src/entities/controllers/file-system-controller-functions/versions.ts

```diff
@@ -0,0 +1,119 @@
+  public static async delete (parentProject : Project, version : ProjectVersionInfo) : Promise<void> {
+    systemConfigurationsStore.items.update(items => {
+      const projectConfigs = items.filter(item => get(item.projectId) === parentProject.identifier);
+      const itemIndex = projectConfigs.findIndex(item => get(item.version) === version.version);
+      items.splice(itemIndex, 1);
+      return items;
+    });
+
+    projectsStore.items.update(projects => {
+      const index = projects.findIndex(project => project.identifier === parentProject.identifier);
+      parentProject.removeVersion(version.version);
+      projects[index] = parentProject;
+      return projects;
+    });
+
+    await ProjectsFileSystemController.update(parentProject);
+    await this.fileApi.deleteVersion(parentProject.toJson(), version.version);
+  }
+
+  private static async import (parentProject : Project, version : string) : Promise<SystemConfiguration> {
+    const versionInfo = await this.fileApi.getVersion(parentProject.projectName, version);
+    return new SystemConfiguration(versionInfo, parentProject.identifier);
+  }
+
+  public static async load (parentProject : Project,  versionString : string) : Promise<void> {
+    const version = await this.import(parentProject, versionString);
+    systemConfigurationsStore.items.update(items => {
+      const index = items.findIndex(item =>
+        get(item.version) === versionString && get(item.projectId) === parentProject.identifier);
+      if(index !== -1) items[index] = new SystemConfigurationStore(version);
+      else items.push(new SystemConfigurationStore(version));
+      return items;
+    });
+  }
+
+
+  public static async loadByProject (project : Project) : Promise<void> {
+    const versionsStrings = project.listVersions();
+    const versions : Array<SystemConfigurationStore> = [];
+    for(const version of versionsStrings) {
+      const versionInfo = await this.import(project, version);
+      versions.push(new SystemConfigurationStore(versionInfo));
+    }
+    systemConfigurationsStore.items.update(items => {
+      return [
+        ...items.filter(item => get(item.projectId) !== project.identifier),
+        ...versions,
+      ];
+    });
+  }
+
+  // Update Functions
+  public static async duplicate (current : ProjectVersionInfo, parentProject : Project) : Promise<void> {
+    const registeredVersions = parentProject.listVersions();
+    const config = await this.fileApi.getVersion(parentProject.projectName, current.version);
+    const highestVersion = registeredVersions.sort((versionA, versionB) => versionB.localeCompare(versionA))[0];
+    const newVersion = this.getNewVersion(highestVersion);
+    const newConfigEntity =
+          new SystemConfiguration({ ...config, version: newVersion }, parentProject.identifier);
+    await this.configuration(parentProject, newConfigEntity);
+  }
+
+  private static getNewVersion (version : string) : string {
```

Comment:
Looking great! Just a note though: This is an utility function, so it looks like something that should be elsewhere, even if it fits in here by the context.

Reply:
Yes, this will be moved in a future PR